app/assets/javascripts/oxalis/model/reducers/settings_reducer.js

```diff
@@ -4,44 +4,73 @@
  */
 
 import _ from "lodash";
-import update from "immutability-helper";
+import {
+  updateKey,
+  updateKey2,
+  updateKey3,
+  type StateShape1,
+  type StateShape2,
+} from "oxalis/model/helpers/deep_update";
 import type { OxalisState } from "oxalis/store";
 import type { ActionType } from "oxalis/model/actions/actions";
 
+//
+// Update helpers
+//
+
+const updateUserConfig = (
+  state: OxalisState,
+  shape: StateShape1<"userConfiguration">,
+): OxalisState => updateKey(state, "userConfiguration", shape);
+
+const updateDatasetConfig = (state: OxalisState, shape: StateShape1<"datasetConfiguration">) =>
+  updateKey(state, "datasetConfiguration", shape);
+
+const updateTemporaryConfig = (state: OxalisState, shape: StateShape1<"temporaryConfiguration">) =>
+  updateKey(state, "temporaryConfiguration", shape);
+
+const updateActiveMapping = (
+  state: OxalisState,
+  shape: StateShape2<"temporaryConfiguration", "activeMapping">,
+) => updateKey2(state, "temporaryConfiguration", "activeMapping", shape);
+
+//
+// Reducer
+//
+
 function SettingsReducer(state: OxalisState, action: ActionType): OxalisState {
   switch (action.type) {
     case "UPDATE_USER_SETTING": {
       const { propertyName, value } = action;
-      return update(state, { userConfiguration: { [propertyName]: { $set: value } } });
+
+      return updateUserConfig(state, { [propertyName]: value });
     }
 
     case "UPDATE_DATASET_SETTING": {
       const { propertyName, value } = action;
-      return update(state, { datasetConfiguration: { [propertyName]: { $set: value } } });
+      return updateDatasetConfig(state, { [propertyName]: value });
     }
 
     case "UPDATE_TEMPORARY_SETTING": {
       const { propertyName, value } = action;
-      return update(state, { temporaryConfiguration: { [propertyName]: { $set: value } } });
+      return updateTemporaryConfig(state, { [propertyName]: value });
     }
 
     case "TOGGLE_TEMPORARY_SETTING": {
       const { propertyName } = action;
-      const value = !state.temporaryConfiguration[propertyName];
-      return update(state, { temporaryConfiguration: { [propertyName]: { $set: value } } });
+      const value: any = !state.temporaryConfiguration[propertyName];
```

Comment:
Is the `: any` left by accident or does it serve a purpose?

Reply:
Unfortunately, it's on purpose. I'll add a comment which explains the reason.